core/src/main/java/io/kestra/core/secret/SecretService.java

```diff
@@ -0,0 +1,17 @@
+package io.kestra.core.secret;
+
+import io.kestra.core.exceptions.IllegalVariableEvaluationException;
+import jakarta.inject.Singleton;
+
+import java.util.Base64;
+import java.util.Optional;
+
+@Singleton
+public class SecretService {
+    public String findSecret(String key) throws IllegalVariableEvaluationException {
+        String environmentVariable = Optional.ofNullable(System.getenv("SECRETS_" + key.toUpperCase()))
```

Comment:
You should get all env var at bean creation time and prepare a map of key/value, this would be more performant than calling System.getenv at each function call.